src/apply.jl

```diff
@@ -339,34 +335,34 @@ function vidal_apply(
     S = replaceind(S, ind_to_replace => ind_to_replace_with')
     V = replaceind(V, ind_to_replace => ind_to_replace_with)
 
-    ψv1, bond_tensors[e], ψv2 = U * Qᵥ₁, S, V * Qᵥ₂
+    ψv1, updated_bond_tensors[e], ψv2 = U * Qᵥ₁, S, V * Qᵥ₂
 
-    for vn in neighbors(ψ, src(e))
+    for vn in neighbors(ψv, src(e))
       if (vn != dst(e))
-        ψv1 = noprime(ψv1 * inv_diag(bond_tensors[vn => src(e)]))
+        ψv1 = noprime(ψv1 * inv_diag(bond_tensor(ψv, vn => src(e))))
       end
     end
 
-    for vn in neighbors(ψ, dst(e))
+    for vn in neighbors(ψv, dst(e))
       if (vn != src(e))
-        ψv2 = noprime(ψv2 * inv_diag(bond_tensors[vn => dst(e)]))
+        ψv2 = noprime(ψv2 * inv_diag(bond_tensor(ψv, vn => dst(e))))
       end
     end
 
     if normalize
       ψv1 /= norm(ψv1)
       ψv2 /= norm(ψv2)
-      normalize!(bond_tensors[e])
+      normalize!(updated_bond_tensors[e])
```

Comment:
```suggestion
      updated_bond_tensors[e] /= norm(updated_bond_tensors[e])
```
Best to avoid in-place updates of tensor data unless there is a compelling reason.